lib/dolly/slugable.rb

```diff
@@ -0,0 +1,61 @@
+module Dolly
+  module Slugable
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+
+    def slug
+      slugable_properties.
+        map(&validate_and_normalize).
+        map(&parameterize_item).
+        join(slugable_separator)
+    end
+
+    def parameterize_item
+      proc do |message|
+        if message.respond_to?(:parameterize)
+          next message.parameterize
+        end
+
+        message
+      end
+    end
+
+    def id
+      doc[:_id] ||= self.class.namespace_key(slug)
+    end
+
+    def validate_and_normalize
+      proc do |property|
+        send(:"#{property}").to_s.tap do |prop|
+          prop.empty? && raise(Dolly::MissingSlugableProperties, [prop])
+        end
+      end
+    end
+
+    module ClassMethods
+      DEFAULT_SEPARATOR = '_'
```

Comment:
Should this be frozen? Or at the frozen string comment at the beginning of the file?

Reply:
added freeze string at bom